x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -292,15 +292,15 @@ public void usageStats(ActionListener<Map<String, Object>> listener) {
     private void addNativeRealms(List<Realm> realms) throws Exception {
         Realm.Factory fileRealm = factories.get(FileRealmSettings.TYPE);
         if (fileRealm != null) {
-            var realmIdentifier = new RealmConfig.RealmIdentifier(FileRealmSettings.TYPE, "default_" + FileRealmSettings.TYPE);
+            var realmIdentifier = new RealmConfig.RealmIdentifier(FileRealmSettings.TYPE, FileRealmSettings.DEFAULT_NAME);
```

Comment:
Given the way realms are configured, you'll have to name the realm if you wish to change any of its settings.
So the `default_file` value is not meant to suggest that other file realms can exist, but that the only file realm can be named differently.

Reply:
Again, not related to your PR, but it's an interesting small topic, so forgive me to drag on this a bit more. I should have mentioned that I found the `default_xxx` names interesting because the reserved realm is just named as `reserved`, which I personally found more concise and less misleading. If we could choose again now, I wonder whether we should not allow users to configure the name of `file` and `native` realms because 1) they don't really do anything; and 2) they create slight confusion; and 3) its a bit awkward when you trying to compare two file realm settings, should they be equal despite having different names?

Reply:
Just want to say this is completely off-topic. So please feel free to ignore it and stay focus on the PR.

Reply:
Good topics.
But, this is not the PR to work on that, and pinning the names of singleton realms is going to be a breaking change and breaking changes are still a burden on customers and I'm not sure it's worth it in this case.